plugins/nf-amazon/src/main/nextflow/cloud/aws/batch/AwsBatchTaskHandler.groovy

```diff
@@ -246,7 +248,8 @@ class AwsBatchTaskHandler extends TaskHandler implements BatchHandler<String,Job
             checkIfUnschedulable(job)
         // fetch the task arn
         if( !taskArn )
-            taskArn = job?.container()?.taskArn()
+            taskArn = getTaskProperties(job)?.taskRoleArn()
+                ?: job?.container()?.taskArn()
```

Comment:
Why now check first `getTaskProperties(job)` and then `job?.container()?.taskArn()` ?

Reply:
I got it from Tom's PR #5391. I think it is needed because of the reuse of job definitions.  If the job was defined with ECS, the information is inside `TaskProperties`. If it is reusing an old definition with container properties, the `taskProperties` are empty and you need to get in the `Container`. It is the same for other job properties. 

Maybe, with the change I did to do not reuse jobs definitions with container properties, it is not needed (https://github.com/nextflow-io/nextflow/pull/6270/commits/54ee92bf69a71dd49f549c79e9f2c0d79c2f3eb1). However, I am not 100% sure because the hash is only applied when container properties are defined, and there could be cases were old definitions are still reused. Maybe the best is just add a comment to explain it.

Reply:
Not sure to follow, if it's an improvement it should go into a separate PR and keep this as a mere SDK upgrade

Reply:
It is not an improvement. When using ECS properties in the task definition (the goal of the PR), some properties that were obtained before from the container are now inside TaskProperties. This is why we check them first in  `getTaskProperties(job)`, it is the expected place with the new definition. We have a fallback to the old container properties because of the job definition reusage. If a job submission is reused with the old container properties, the task properties will be empty and we should get them from the container properties as before.

Reply:
ok, triggered e2e tests